extensions/ql-vscode/src/common/disposable-object.ts

```diff
@@ -88,3 +88,7 @@ export class DisposableObject implements Disposable {
     }
   }
 }
+
+export function isDisposable(obj: unknown): obj is Disposable {
+  return obj !== undefined && typeof (obj as Disposable).dispose === "function";
```

Comment:
I believe this will throw an error if `obj` is `null`.

You could add a `obj !== null` check, but you could write it without using `as` at all like:
```typescript
return (
  obj !== null &&
  typeof obj === "object" &&
  "dispose" in obj &&
  typeof obj.dispose === "function"
);
```